gravitee-apim-gateway/gravitee-apim-gateway-tests-sdk/src/main/java/io/gravitee/apim/gateway/tests/sdk/container/GatewayTestContainer.java

```diff
@@ -33,26 +33,48 @@
 import io.gravitee.tracing.api.Tracer;
 import io.vertx.core.Vertx;
 import java.util.List;
+import java.util.function.Consumer;
 import org.mockito.Mockito;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 /**
- * This class allows to extends the {@link GatewayContainer} in order to be able to override the {@link NodeFactory}
+ * This class allows to extend the {@link GatewayContainer} in order to override the {@link NodeFactory}
  * and provide a {@link GatewayTestNode}
  *
  * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
  * @author GraviteeSource Team
  */
 public class GatewayTestContainer extends GatewayContainer {
 
+    private final Consumer<ApplicationContext> onBootApplicationContextCreated;
+
+    /**
+     * Constructor with action to execute once the boot application context has been created.
+     * This allows for setting up specific requirements for testing (e.g. customizing gravitee yaml configuration, registering boot plugins, ...)
+     *
+     * @param onBootApplicationContextCreated actions to execute once Spring context is created and before starting components.
+     */
+    public GatewayTestContainer(Consumer<ApplicationContext> onBootApplicationContextCreated) {
```

Comment:
`@RequiredArgsConstructor` ?

Reply:
I wanted to add a Javadoc to explain the purpose of that constructor so I have avoided the use of `@RequiredArgsConstructor`